TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -532,6 +532,18 @@ public override bool Visit(Program program)
             CheckEndProgram(program);
             
             FormalizedCommentsChecker.CheckProgramComments(program);
+            if (program.IsNested)
+            {  // Check that a Nested Program cannot have a CONFIGURATION SECTION
+                IList<EnvironmentDivision> envDiv = program.GetChildren<EnvironmentDivision>();
+                if (envDiv.Count > 0)
+                {
+                    IList<ConfigurationSection> confSection = envDiv[0].GetChildren<ConfigurationSection>();
+                    if (confSection.Count > 0)
+                    {
+                        DiagnosticUtils.AddError(confSection[0], "A Nested Program cannot have a CONFIGURATION SECTION.");
+                    }
+                }
+            }
```

Comment:
I'd rather override `Visit(ConfigurationSection configurationSection)` method and then use `GetProgramNode` to access the declaring program. This would avoid iterating over children and test type twice.